src/openvic-simulation/utility/Deque.hpp

```diff
@@ -0,0 +1,1910 @@
+// deque standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#if __has_include(<__msvc_string_view.hpp>)
+
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#include <xmemory>
+
+#if _HAS_CXX17
+#include <xpolymorphic_allocator.h>
+#endif // _HAS_CXX17
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+namespace OpenVic::utility::detail::__deque {
+	using namespace ::std;
+
+	template<class _Mydeque>
+	class _Deque_unchecked_const_iterator {
+	private:
+		using _Size_type = typename _Mydeque::size_type;
+
+		static constexpr int _Block_size = _Mydeque::_Block_size;
+
+	public:
+		using iterator_category = random_access_iterator_tag;
+
+		using value_type = typename _Mydeque::value_type;
+		using difference_type = typename _Mydeque::difference_type;
+		using pointer = typename _Mydeque::const_pointer;
+		using reference = const value_type&;
```

Comment:
```suggestion
		using reference = value_type const&;
```
Non-standard `const&` and `const*` formatting here and throughout this file

Reply:
I'm trying to leave it as default to the MSVC STL as possible.